src/lib/containers/synapse_table_functions/SynapseTableCell.tsx

```diff
@@ -77,150 +46,152 @@ export const SynapseTableCell: React.FC<SynapseTableCellProps> = ({
-        columnName={columnName}
-        selectColumns={selectColumns}
-        columnModels={columnModels}
-        isHeader={false}
-        labelLink={columnLinkConfig}
-        rowData={rowData}
-      />
-    )
-  }
-  if (
-    entityColumnIndicies.includes(colIndex) &&
-    Object.prototype.hasOwnProperty.call(mapEntityIdToHeader, columnValue)
-  ) {
-    return (
-      <EntityLink
-        entityHeader={mapEntityIdToHeader[columnValue]}
-        className={`${isBold}`}
-      />
-    )
-  }
-  if (dateListColumnIndicies.includes(colIndex)) {
-    const jsonData: number[] = JSON.parse(columnValue)
-    return (
-      <p>
-        {jsonData.map((val: number, index: number) => {
-          return (
-            <span key={index} className={isBold}>
-              {new Date(val).toLocaleString()}
-              {index !== jsonData.length - 1 ? ', ' : ''}
-            </span>
-          )
-        })}{' '}
-      </p>
-    )
-  }
-  if (booleanListColumnIndicies.includes(colIndex)) {
-    const jsonData: boolean[] = JSON.parse(columnValue)
-    return (
-      <p>
-        {jsonData.map((val: boolean, index: number) => {
-          return (
-            <span key={index} className={isBold}>
-              {val ? 'true' : 'false'}
-              {index !== jsonData.length - 1 ? ', ' : ''}
-            </span>
-          )
-        })}
-      </p>
-    )
-  }
-
-  // If it's a file from a table view
-  if (fileHandleIdColumnIndicies.includes(colIndex)) {
-    return (
-      <>
-        <DirectDownload
-          associatedObjectId={tableEntityId!}
-          associatedObjectType={FileHandleAssociateType.TableEntity}
-          fileHandleId={columnValue}
-          displayFileName={true}
-        />
-      </>
-    )
-  }
-
-  // If it's a list of entity ids
-  if (entityIdListColumnIndicies.includes(colIndex)) {
-    const jsonData: string[] = JSON.parse(columnValue)
-    return <EntityIdList entityIdList={jsonData} />
-  }
 
-  if (otherListColumnIndicies.includes(colIndex)) {
-    const jsonData: string[] = JSON.parse(columnValue)
-    return (
-      <p>
-        {jsonData.map((val: string, index: number) => {
-          return (
-            <span key={val} className={isBold}>
-              {val}
-              {index !== jsonData.length - 1 ? ', ' : ''}
-            </span>
-          )
-        })}
-      </p>
-    )
-  }
-  if (dateColumnIndicies.includes(colIndex)) {
-    return (
-      <p className={isBold}>{new Date(Number(columnValue)).toLocaleString()}</p>
-    )
-  } else if (
-    userColumnIndicies.includes(colIndex) &&
-    Object.prototype.hasOwnProperty.call(mapUserIdToHeader, columnValue)
-  ) {
-    const { ownerId, userName } = mapUserIdToHeader[columnValue]
-    if (mapUserIdToHeader[columnValue].isIndividual === false) {
-      // isUserGroupHeader
-      const icon = userName === AUTHENTICATED_USERS ? 'globe-americas' : 'users'
-      if (userName === AUTHENTICATED_USERS) {
+  switch (columnType) {
```

Comment:
The cases should be in the same order as the if blocks. I like this a lot more because we can more quickly see how we are (or aren't) handling certain column types.